crates/valence_protocol/src/lib.rs

```diff
@@ -118,9 +118,23 @@ mod var_long;
 /// Used only by proc macros. Not public API.
 #[doc(hidden)]
 pub mod __private {
+    use std::marker::PhantomData;
+
     pub use anyhow::{anyhow, bail, ensure, Context, Result};
 
     pub use crate::{Decode, DecodePacket, Encode, EncodePacket, VarInt};
+
+    impl<T> Encode for PhantomData<T> {
+        fn encode(&self, _: impl std::io::Write) -> Result<()> {
+            Ok(())
+        }
+    }
+
+    impl<'a, T> Decode<'a> for PhantomData<T> {
+        fn decode(_: &mut &'a [u8]) -> Result<Self> {
+            Ok(PhantomData)
+        }
+    }
```

Comment:
what is this for?

Reply:
Sorry, unrelated. There was an attempt to make `Proterty<S>` has Cow<'_, str> as a default, but I passed for now